diff --git a/node_modules/react-native-raw-bottom-sheet/index.d.ts b/node_modules/react-native-raw-bottom-sheet/index.d.ts
index 31a5259..d8434b6 100644
--- a/node_modules/react-native-raw-bottom-sheet/index.d.ts
+++ b/node_modules/react-native-raw-bottom-sheet/index.d.ts
@@ -20,6 +20,7 @@ declare module "react-native-raw-bottom-sheet" {
       draggableIcon?: StyleProp<ViewStyle>;
     };
     keyboardAvoidingViewEnabled?: boolean;
+    closeOnDragAboveSheet?: boolean;
   };
 
   export default class RBSheet extends Component<RBSheetProps> {
diff --git a/node_modules/react-native-raw-bottom-sheet/src/index.js b/node_modules/react-native-raw-bottom-sheet/src/index.js
index 0b3eab0..db0317c 100644
--- a/node_modules/react-native-raw-bottom-sheet/src/index.js
+++ b/node_modules/react-native-raw-bottom-sheet/src/index.js
@@ -25,7 +25,8 @@ class RBSheet extends Component {
     this.state = {
       modalVisible: false,
       animatedHeight: new Animated.Value(0),
-      pan: new Animated.ValueXY()
+      pan: new Animated.ValueXY(),
+      isShown: false
     };
 
     this.createPanResponder(props);
@@ -37,10 +38,10 @@ class RBSheet extends Component {
     if (visible) {
       this.setState({ modalVisible: visible });
       if (typeof onOpen === "function") onOpen(props);
-      Animated.timing(animatedHeight, {
+      Animated.spring(animatedHeight, {
         useNativeDriver: false,
         toValue: height,
-        duration: openDuration
+        duration: 0
       }).start();
     } else {
       Animated.timing(animatedHeight, {
@@ -96,9 +97,10 @@ class RBSheet extends Component {
       closeOnPressBack,
       children,
       customStyles,
-      keyboardAvoidingViewEnabled
+      keyboardAvoidingViewEnabled,
+      closeOnDragAboveSheet,
     } = this.props;
-    const { animatedHeight, pan, modalVisible } = this.state;
+    const { animatedHeight, pan, modalVisible, isShown } = this.state;
     const panStyle = {
       transform: pan.getTranslateTransform()
     };
@@ -112,6 +114,7 @@ class RBSheet extends Component {
         onRequestClose={() => {
           if (closeOnPressBack) this.setModalVisible(false);
         }}
+        onShow={()=> this.setState({isShown:true})}
       >
         <KeyboardAvoidingView
           enabled={keyboardAvoidingViewEnabled}
@@ -123,7 +126,7 @@ class RBSheet extends Component {
             activeOpacity={1}
             onPress={() => (closeOnPressMask ? this.close() : null)}
           />
-          <Animated.View
+         {isShown && <Animated.View 
             {...(!dragFromTopOnly && this.panResponder.panHandlers)}
             style={[panStyle, styles.container, { height: animatedHeight }, customStyles.container]}
           >
@@ -132,11 +135,11 @@ class RBSheet extends Component {
                 {...(dragFromTopOnly && this.panResponder.panHandlers)}
                 style={styles.draggableContainer}
               >
-                <View style={[styles.draggableIcon, customStyles.draggableIcon]} />
+                <View style={[styles.draggableIcon,closeOnDragAboveSheet && styles.draggableIconAboveSheet, customStyles.draggableIcon,]} />
               </View>
             )}
             {children}
-          </Animated.View>
+          </Animated.View>}
         </KeyboardAvoidingView>
       </Modal>
     );
@@ -157,7 +160,8 @@ RBSheet.propTypes = {
   customStyles: PropTypes.objectOf(PropTypes.object),
   onClose: PropTypes.func,
   onOpen: PropTypes.func,
-  children: PropTypes.node
+  children: PropTypes.node,
+  closeOnDragAboveSheet: PropTypes.bool,
 };
 
 RBSheet.defaultProps = {
@@ -174,7 +178,8 @@ RBSheet.defaultProps = {
   customStyles: {},
   onClose: null,
   onOpen: null,
-  children: <View />
+  children: <View />,
+  closeOnDragAboveSheet: false
 };
 
 export default RBSheet;
diff --git a/node_modules/react-native-raw-bottom-sheet/src/style.js b/node_modules/react-native-raw-bottom-sheet/src/style.js
index a70d1fe..bb09258 100644
--- a/node_modules/react-native-raw-bottom-sheet/src/style.js
+++ b/node_modules/react-native-raw-bottom-sheet/src/style.js
@@ -13,7 +13,7 @@ const styles = StyleSheet.create({
     backgroundColor: "#fff",
     width: "100%",
     height: 0,
-    overflow: "hidden"
+    // overflow: "hidden"
   },
   draggableContainer: {
     width: "100%",
@@ -26,6 +26,14 @@ const styles = StyleSheet.create({
     borderRadius: 5,
     margin: 10,
     backgroundColor: "#ccc"
+  },
+  draggableIconAboveSheet:{
+    width: 35,
+    height: 5,
+    borderRadius: 5,
+    margin: 10,
+    position:'absolute',
+    top:-22
   }
 });
 
