diff --git a/node_modules/react-native-video-controls/VideoPlayer.js b/node_modules/react-native-video-controls/VideoPlayer.js
index 7927a5c..87878b7 100644
--- a/node_modules/react-native-video-controls/VideoPlayer.js
+++ b/node_modules/react-native-video-controls/VideoPlayer.js
@@ -549,10 +549,10 @@ export default class VideoPlayer extends Component {
    * or duration. Formatted to look as 00:00.
    */
   calculateTime() {
-    if (this.state.showTimeRemaining) {
-      const time = this.state.duration - this.state.currentTime;
-      return `-${this.formatTime(time)}`;
-    }
+    // if (this.state.showTimeRemaining) {
+    //   const time = this.state.duration - this.state.currentTime;
+    //   return `-${this.formatTime(time)}`;
+    // }
 
     return this.formatTime(this.state.currentTime);
   }
@@ -1035,7 +1035,7 @@ export default class VideoPlayer extends Component {
           {...this.player.volumePanResponder.panHandlers}>
           <Image
             style={styles.volume.icon}
-            source={require('./assets/img/volume.png')}
+            source={(this.state.volume === 0) ? require('./assets/img/volume_mute.png') : require('./assets/img/volume.png')}
           />
         </View>
       </View>
@@ -1122,7 +1122,7 @@ export default class VideoPlayer extends Component {
               styles.seekbar.fill,
               {
                 width: this.state.seekerFillWidth,
-                backgroundColor: this.props.seekColor || '#FFF',
+                backgroundColor: this.props.seekColor || '#333',
               },
             ]}
             pointerEvents={'none'}
